drivers/serial/serial.c

```diff
@@ -1647,21 +1653,25 @@ int uart_register(FAR const char *path, FAR uart_dev_t *dev)
   dev->pid = INVALID_PROCESS_ID;
 #endif
 
-#ifdef CONFIG_SERIAL_TERMIOS
   /* If this UART is a serial console */
 
   if (dev->isconsole)
     {
+#ifdef CONFIG_SERIAL_TERMIOS
       /* Enable signals by default */
 
       dev->tc_lflag |= ISIG;
 
       /* Enable \n -> \r\n translation for the console */
 
       dev->tc_oflag = OPOST | ONLCR;
-    }
 #endif
 
+      /* Convert CR to LF by default for console */
+
+      dev->tc_iflag |= ICRNL;
```

Comment:
let's follow how tc_oflag get processed specially for console in case of CONFIG_SERIAL_TERMIOS isn't enable.